src/utils/uploads.js

```diff
@@ -243,6 +259,35 @@ async function getFileFromDataTransferItem(
     return file;
 }
 
+/**
+ * Get file from DataTransferItem or UploadDataTransferItemWithAPIOptions
+ * Uses `entry`'s `getAsFile` method for retrieving package information as a single file.
+ * @see https://en.wikipedia.org/wiki/Package_(macOS)
+ *
+ * @param {UploadDataTransferItemWithAPIOptions | DataTransferItem} itemData
+ * @returns {Promise<?UploadFile | ?UploadFileWithAPIOptions | null>}
+ */
+async function getPackageFileFromDataTransferItem(
+    itemData: UploadDataTransferItemWithAPIOptions | DataTransferItem,
+): Promise<?UploadFile | ?UploadFileWithAPIOptions | null> {
+    const item = getDataTransferItem(itemData);
+    const entry = getEntryFromDataTransferItem(((item: any): DataTransferItem));
+    if (!entry) {
+        return null;
+    }
+
+    const itemFile = item.getAsFile();
+
+    if (doesDataTransferItemContainAPIOptions(itemData)) {
+        return Promise.resolve({
+            file: ((itemFile: any): UploadFile),
+            options: getDataTransferItemAPIOptions(itemData),
+        });
+    }
+
+    return Promise.resolve(itemFile);
```

Comment:
Are these `Promise.resolve` required?

Reply:
They may not be. it has been a bit since I added this in, so let me look again.

Reply:
@ConradJChan I made this one return promises like this b/c the associated functions: `getFileFromEntry`, `getFileFromDataTransferItem` are all async. This new function already has all the info it needs, but I wanted to guard against introducing a non-async function along side the others, so we wouldn't have a mismatch in return types anywhere. 

There may be a tidier way to write these return values, e.g., having a single `Promise.resolve` with the function's logic in there.